indico/queries/teach_tasks.py

```diff
@@ -0,0 +1,244 @@
+        super().__init__(
+            query=self.query,
+            variables={
+                "labels": labels,
+                "dataset_id": dataset_id,
+                "labelset_id": labelset_id,
+            },
+        )
+
+
+class GetQuestionaireExamples(GraphQLRequest):
+    """
+    Gets unlabeled examples from a Questionnaire.
+
+    Args:
+        questionaire_id (int): The id of the questionnaire to get examples from.
+        num_examples (int): The number of examples to get from the questionaire.
+
+    Returns:
+        List[Example]
+
+    """
+
+    query = """
+    query(
+        $questionaire_id: Int!,
+        $num_examples: Int!
+    )
+    {
+        questionnaires(questionnaireIds: [$questionaire_id]) {
+            questionnaires {
+                examples(numExamples: $num_examples) {
+                    rowIndex
+                    datafileId
+                    source
+                }
+            }
+        }
+    }
+    """
+
+    def __init__(self, questionaire_id: int, num_examples: int):
+        super().__init__(
+            query=self.query,
+            variables={
+                "questionaire_id": questionaire_id,
+                "num_examples": num_examples,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            examples = [
+                Example(**e)
+                for e in super().process_response(response)["questionnaires"][
+                    "questionnaires"
+                ][0]["examples"]
+            ]
+        except IndexError:
+            raise IndicoNotFound(
+                "Examples not found. Please check the ID you are using."
+            )
+        return examples
+
+
+class CreateQuestionaire(GraphQLRequest):
+    """
+    Creates the questionnaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        dataset_id (int): The id of the dataset to create the questionnaire from.
+        source_column_id (int): The id of the source column to create a questionnaire from.
+        targets (List[str]): The classes for the dataset.
+
+    Returns:
+        Questionnaire
+
+    """
+
+    query = """
+        mutation(
+            $name: String!,
+            $dataset_id: Int!,
+            $questions: [QuestionInput]!,
+            $source_col_id: Int!,
+        ) {
+            createQuestionnaire (
+                datasetId: $dataset_id,
+                dataType: TEXT,
+                name: $name,
+                numLabelersRequired: 1,
+                questions: $questions,
+                sourceColumnId: $source_col_id,
+                instructions: ""
+            ) {
+                id
+            }
+        }
+    """
+
+    def __init__(
+        self, name: str, dataset_id: int, source_column_id: int, targets: List[str]
+    ):
+        questions = [
+            {"type": "ANNOTATION", "targets": targets, "keywords": [], "text": name,}
```

Comment:
So we are only allowing annotation tasks? We can add an arg to the query for the type of question.